### PR TITLE
Update dependencies, use native node `fetch`, drop Node<18 for some packages

### PR DESCRIPTION
**Pull request by askoufis**
- Updated various dependencies
- Removed some unused dependencies
- Add an `engines` config for `@vocab/phrase` and `@vocab/cli`, make `@vocab/phrase` use native fetch instead of `node-fetch`

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -15,22 +15,22 @@
...
     "fast-glob": "^3.2.4",
     "form-data": "^4.0.0",
-    "node-fetch": "^2.6.1",
-    "prettier": "^2.1.2",
-    "yargs": "^16.1.0"
+    "yargs": "^17.7.2"
   },
   "devDependencies": {
-    "@types/node-fetch": "^2.5.7",
-    "@types/prettier": "^2.1.5",
```
Comment (askoufis) [inline packages/cli/package.json:33]: `prettier@3` comes with types inbuilt.

File: packages/cli/package.json [hunk 2]
```diff
@@ -15,22 +15,22 @@
     "dist",
     "bin.js"
   ],
+  "engines": {
+    "node": ">=18"
+  },
   "author": "SEEK",
   "license": "MIT",
   "dependencies": {
-    "@types/env-ci": "^3.1.0",
```
Comment (askoufis) [inline packages/cli/package.json:21]: Should've been a dev dep in the first place.

File: packages/phrase/src/phrase-api.ts [hunk 3]
```diff
@@ -161,7 +160,7 @@ export async function pushTranslations(
       | undefined
     >(`uploads`, {
       method: 'POST',
-      body: formData,
+      body: formData.toString(),
```
Comment (askoufis) [inline packages/phrase/src/phrase-api.ts:163]: `node-fetch`'s `body` type accepts the type of `formData`, but native fetch's doesn't, so converting to string.

File: test-helpers/src/helpers.ts [hunk 4]
```diff
@@ -57,7 +57,9 @@ export const runServerFixture = (
     await compileFixtureTranslations(fixtureName);
 
     const port = portCounter++;
-    const getConfig = require(`@vocab-fixtures/${fixtureName}/webpack.config.js`);
+    const getConfig = require(
+      `@vocab-fixtures/${fixtureName}/webpack.config.js`,
+    );
```
Comment (askoufis) [inline test-helpers/src/helpers.ts:62]: `prettier@3` format.

File: packages/core/src/compile.ts [hunk 5]
```diff
@@ -236,7 +236,7 @@ export async function generateRuntime(loadedTranslation: LoadedTranslation) {
     imports,
     loadedTranslation,
   );
-  const declaration = prettier.format(serializedTranslationType, {
+  const declaration = await prettier.format(serializedTranslationType, {
```
Comment (askoufis) [inline packages/core/src/compile.ts:239]: All `prettier` APIs are now async.

File: packages/cli/package.json [hunk 6]
```diff
@@ -15,22 +15,22 @@
...
     "@vocab/core": "workspace:^",
     "@vocab/phrase": "workspace:^",
     "env-ci": "^7.3.0",
     "fast-glob": "^3.2.4",
     "form-data": "^4.0.0",
-    "node-fetch": "^2.6.1",
-    "prettier": "^2.1.2",
```
Comment (askoufis) [inline packages/cli/package.json:28]: The CLI doesn't use `prettier`.

File: packages/phrase/package.json [hunk 7]
```diff
@@ -15,12 +18,11 @@
     "csv-stringify": "^6.2.3",
     "debug": "^4.3.1",
     "form-data": "^4.0.0",
-    "node-fetch": "^2.6.1",
     "picocolors": "^1.0.0"
   },
   "devDependencies": {
     "@types/debug": "^4.1.5",
-    "@types/node-fetch": "^2.5.7"
+    "@types/node": "^20.11.22"
```
Comment (michaeltaranto) [inline packages/phrase/package.json:25]: If the `engines` config specifies 18, should the types be set to 18 to ensure you dont rely on v20 features accidentally?

Comment (askoufis) [reply packages/phrase/package.json:25]: Not sure on this one. Ideally we'd do a matrix test on CI to test all LTS versions above 18 or something, but we barely use anything modern anyway, other than `fetch`. Happy to go either way though, I'm not against downgrading.

Comment (askoufis) [reply packages/phrase/package.json]: Downgrade to node 18

**Review by michaeltaranto — APPROVED** [review]

**Pull request merged**